### PR TITLE
feat: update for clone over UI

### PR DESCRIPTION
**Pull request by harshithmullapudi**
## What
closes https://github.com/airbytehq/airbyte-internal-issues/issues/366

## It has 2 features
- [x] API support for taking name while cloning
- [x] UI feature to clone the source

### CODE REVIEW

File: airbyte-webapp/src/config/defaultConfig.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ const defaultConfig: Config = {
   segment: { enabled: true, token: "" },
   healthCheckInterval: 20000,
   version: "dev",
-  apiUrl: `${window.location.protocol}//${window.location.hostname}:8001/api/v1/`,
```
Comment (timroes) [inline airbyte-webapp/src/config/defaultConfig.ts:23]: Development change left over. To only overwrite this in development, just create a file called `.env.development` in `airbyte-webapp` and put the following into it:

```
REACT_APP_API_URL=https://...
```

**Comment by timroes** [comment]
We currently have a refactoring on the query infrastructure ongoing which will collide with that. I am sorry, but I feel we need to put this on hold shortly, until we've merged the other PR, and it will require some more adjustments to this PR after it's been merged.

**Comment by harshithmullapudi** [comment]
Sure @timroes can you help with the waiting PR so that I can track it

**Comment by timroes** [comment]
> Sure @timroes can you help with the waiting PR so that I can track it

I am not sure I get what you're meaning by that? What waiting PR exactly are you refering to?

**Comment by timroes** [comment]
@harshithmullapudi We've now merged the above mentioned refactoring and have removed `rest-hooks` from the code. You should now be able to bring this PR up-to-date on the react-query infrastructure.

**Comment by harshithmullapudi** [comment]
Sure will start working on the PR

**Comment by cgardens** [comment]
@harshithmullapudi removing myself as a reviewer to reduce notification noise. Feel free to re-add me when this is ready for review.

**Pull request closed without merging**